module/module.go

```diff
@@ -506,6 +514,63 @@ func (m *Module) AddResource(ctx context.Context, req *pb.AddResourceRequest) (*
 	return &pb.AddResourceResponse{}, nil
 }
 
+// DiscoverComponents takes a list of discovery queries and returns corresponding
+// component configurations.
+func (m *Module) DiscoverComponents(
+	ctx context.Context,
+	req *robotpb.DiscoverComponentsRequest,
+) (*robotpb.DiscoverComponentsResponse, error) {
+	var discoveries []*robotpb.Discovery
+
+	for _, q := range req.Queries {
+		// Handle triplet edge case i.e. if the subtype doesn't contain ':', add the "rdk:component:" prefix
+		if !strings.ContainsRune(q.Subtype, ':') {
+			q.Subtype = "rdk:component:" + q.Subtype
+		}
```

Comment:
i _think_ triplets are used/supported throughout now, so this guard might not be necessary here and in the corresponding version in robot/server/server.go.

anyway, not necessary to figure this out now since it doesn't block your change - we can follow up on this.

Reply:
gotcha thanks for context

Added to maintain somewhat parallel logic to the server.go version

Reply:
> Added to maintain somewhat parallel logic to the server.go version

yup that's the right move - this should be in parity until we (netcode) address the logic in the robot server version of this function